src/core/ext/xds/xds_audit_logger_registry.cc

```diff
@@ -0,0 +1,100 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include <grpc/support/port_platform.h>
+
+#include "src/core/ext/xds/xds_audit_logger_registry.h"
+
+#include <utility>
+
+#include "absl/strings/string_view.h"
+#include "absl/types/optional.h"
+#include "absl/types/variant.h"
+#include "envoy/config/core/v3/extension.upb.h"
+#include "envoy/config/rbac/v3/rbac.upb.h"
+
+#include "src/core/ext/xds/xds_common_types.h"
+#include "src/core/lib/gprpp/validation_errors.h"
+#include "src/core/lib/json/json.h"
+
+namespace grpc_core {
+
+namespace {
+
+class StdoutLoggerConfigFactory : public XdsAuditLoggerRegistry::ConfigFactory {
+ public:
+  Json::Object ConvertXdsAuditLoggerConfig(
+      const XdsResourceType::DecodeContext& /*context*/,
+      absl::string_view /*configuration*/,
+      ValidationErrors* /*errors*/) override {
+    return Json::Object{{"stdout_logger", Json::Object()}};
+  }
+
+  absl::string_view type() override { return Type(); }
+
+  static absl::string_view Type() {
+    return "envoy.extensions.rbac.audit_loggers.stream.v3.StdoutAuditLog";
+  }
+};
+
+}  // namespace
+
+XdsAuditLoggerRegistry::XdsAuditLoggerRegistry() {
+  audit_logger_config_factories_.emplace(
+      StdoutLoggerConfigFactory::Type(),
+      std::make_unique<StdoutLoggerConfigFactory>());
+}
+
+Json XdsAuditLoggerRegistry::ConvertXdsAuditLoggerConfig(
+    const XdsResourceType::DecodeContext& context,
+    const envoy_config_rbac_v3_RBAC_AuditLoggingOptions_AuditLoggerConfig*
+        logger_config,
+    ValidationErrors* errors) const {
+  const auto* typed_extension_config =
+      envoy_config_rbac_v3_RBAC_AuditLoggingOptions_AuditLoggerConfig_audit_logger(
+          logger_config);
+  // It is okay if this is not present.
+  if (typed_extension_config == nullptr) {
+    return Json();  // A null Json object.
+  } else {
+    ValidationErrors::ScopedField field(errors,
+                                        ".typed_extension_config.typed_config");
+    const auto* typed_config =
+        envoy_config_core_v3_TypedExtensionConfig_typed_config(
+            typed_extension_config);
+    auto extension = ExtractXdsExtension(context, typed_config, errors);
+    if (!extension.has_value()) return Json();
+    // Check for registered audit logger type.
+    absl::string_view* serialized_value =
+        absl::get_if<absl::string_view>(&extension->value);
+    if (serialized_value != nullptr) {
+      auto config_factory_it =
+          audit_logger_config_factories_.find(extension->type);
+      if (config_factory_it != audit_logger_config_factories_.end()) {
+        return config_factory_it->second->ConvertXdsAuditLoggerConfig(
```

Comment:
Before we return here, we need to validate the returned config against the gRPC audit logger registry.  (This can be a TODO for now if this gets merged before the other PR.)

Reply:
Added a TODO. If the API PR is merged first, I will rebase and remove it.